demo/full/scripts/controllers/Settings.jsx

```diff
@@ -0,0 +1,259 @@
+import React from "react";
+
+import { getCheckBoxValue } from "./ContentList";
+
+import Option from "../components/Options/Option";
+import Playback from "../components/Options/Playback.jsx";
+import InitSettings from "../components/Options/InitSettings.jsx";
+import NetworkConfig from "../components/Options/NetworkConfig.jsx";
+import TrackSwitch from "../components/Options/TrackSwitch.jsx";
+import BufferOptions from "../components/Options/BufferOptions.jsx";
+
+class Settings extends React.Component {
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      autoPlay: true,
+      manualBrSwitchingMode: "seamless",
+      initialVideoBr: 0,
+      initialAudioBr: 0,
+      minVideoBr: 0,
+      minAudioBr: 0,
+      maxVideoBr: Infinity,
+      maxAudioBr: Infinity,
+      limitVideoWidth: false,
+      throttleVideoBitrateWhenHidden: false,
+      stopAtEnd: true,
+      segmentRetry: 4,
+      manifestRetry: 4,
+      offlineRetry: Infinity,
+      enableFastSwitching: true,
+      audioTrackSwitchingMode: "seamless",
+      onCodecSwitch: "continue",
+      wantedBufferAhead: 30,
+      maxBufferAhead: Infinity,
+      maxBufferBehind: Infinity,
+    };
+  }
+
+  getOptions = () => {
```

Comment:
I don't understand this syntax.

What does it means? Why not just `getOptions() { /* ... */ }`?

Reply:
Yes, of course, I forgot to remove and switch to a more "classic" way.

However, sometimes those arrow functions are useful when u want to omit the `.bind(this)` call, cause the function is losing the context when u pass to another component (function).

Reply:
Oh ok, so that's basically arrow function as a property, in which case the context of `this` is always refering to the class if I understand correctly?